examples/apigee-x-vpn-hub-and-spoke-design/apigee_x_project.tf

```diff
@@ -0,0 +1,58 @@
+##########################################################
+# Copyright 2021 Google LLC.
+# This software is provided as-is, without warranty or
+# representation for any use or purpose.
+# Your use of it is subject to your agreement with Google.
+#
+# Sample Terraform script to set up an Apigee X instance 
+##########################################################
+
+##########################################################
+### Create a Google Cloud Project for Apigee X runtime
+##########################################################
+
+resource "google_project" "apigee_x_project" {
```

Comment:
You can use https://github.com/terraform-google-modules/terraform-google-project-factory which has some of the features (like random suffix), service enablement etc built in